File: android/app/src/main/kotlin/com/example/hotel_reception/procesor/model/SetupDto.kt
//package com.example.hotel_reception.procesor.model
//
//import org.json.JSONObject
//
//class SetupDto(val roomsByType: List<RoomCount>, val service: ServiceDto) {
//    companion object {
//        fun fromJson(json: String): SetupDto {
//            val data = JSONObject(json)
//            val service = data["service"].toString().toServiceType()
//            val roomsList = (data["roomsByType"] as List<*>).map {
//                val roomByType = JSONObject(it.toString())
//                val type = roomByType["type"].toString().toRoomType()
//                val count = roomByType["count"].toString().toIntOrNull() ?: 0
//                return@map RoomCount(type, count)
//            }
//            return SetupDto(roomsList, service)
//        }
//    }
//
//}
//
//data class RoomCount(val type: RoomTypeDto, val count: Int)
//
//enum class ServiceDto {
//    normal,
//    medium,
//    expensive,
//}
//
//fun String.toServiceType(): ServiceDto {
//    return when (this) {
//        ServiceDto.normal.name -> ServiceDto.normal
//        ServiceDto.medium.name -> ServiceDto.medium
//        ServiceDto.expensive.name -> ServiceDto.expensive
//        else -> throw Exception("Unsupported type $this")
//    }
//}
//
//enum class RoomTypeDto {
//    econom,
//    suite,
//    president,
//}
//
//fun String.toRoomType(): RoomTypeDto {
//    return when (this) {
//        RoomTypeDto.econom.name -> RoomTypeDto.econom
//        RoomTypeDto.suite.name -> RoomTypeDto.suite
//        RoomTypeDto.president.name -> RoomTypeDto.president
//        else -> throw Exception("Unsupported type $this")
//    }
//}

File: lib/model/dto/setup_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:hotel_reception/model/dto/room_count_dto.dart';
import 'package:hotel_reception/model/service.dart';

part 'setup_dto.freezed.dart';
part 'setup_dto.g.dart';

@freezed
class SetupDto with _$SetupDto {
  factory SetupDto({
    required List<RoomCount> roomsByType,
    required Service service,
    @Default(1) int dayTime,
  }) = _SetupDto;

  factory SetupDto.fromJson(Map<String, dynamic> json) =>
      _$SetupDtoFromJson(json);
}
